### PR TITLE
More rules for common CMPs

### PR DESCRIPTION
**Pull request by muodov**
[Openli (Legalmonster) CMP](https://app.asana.com/0/1203268166580279/1203839080310161/f)
[cookieconsent CMP](https://app.asana.com/0/1203268166580279/1206134345039234/f)
[cookieAcceptBar notice](https://app.asana.com/0/1203268166580279/1206408867217747/f)
[hu-manity.co CMP](https://app.asana.com/0/1203268166580279/1206336915285063/f)
[Termsfeed CMP](https://app.asana.com/0/1203268166580279/1202850714857134/f)
[Taunton CMP](https://app.asana.com/0/1203268166580279/1206012592804743/f)
[Cookie Popup on pabcogypsum.com](https://app.asana.com/0/1203268166580279/1206399569931771/f)
[CMP on www.abconcerts.be and others](https://app.asana.com/0/1203268166580279/1205767713093504/f)
[OM Cookie Manager](https://app.asana.com/0/1203268166580279/1206115145833772/f)
[Lineagrafica CMP](https://app.asana.com/0/1203268166580279/1204727226081786/f)

### CODE REVIEW

File: rules/autoconsent/openli.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  "vendorUrl": "https://openli.com",
+  "prehideSelectors": [".legalmonster-cleanslate"],
+  "detectCmp": [
+    {
+      "exists": ".legalmonster-cleanslate"
+    }
+  ],
+  "detectPopup": [
+    {
+      "visible": ".legalmonster-cleanslate",
```
Comment (sammacbeth) [inline rules/autoconsent/openli.json:12]: This seems to be too permissive, and leading to the rule triggering on every page load. On chargebee.com, this matches the shield in the bottom left corner which is always visible:
Screenshot 2024-02-01 at 13 21 42

File: tests/om.spec.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import generateCMPTests from "../playwright/runner";
+
+generateCMPTests('om', [
+  'https://olli-machts.de/en/extension/cookie-manager',
+  'https://www.uni-assist.de/',
+  'https://www.unisabana.edu.co/',
+  'https://deutsche-alzheimer.de/',
```
Comment (sammacbeth) [inline tests/om.spec.ts:7]: Testing on these websites I saw the following issues:
1. The popup comes back on every page load. This even happens when I do the opt-out manually, so seems to be an issue with the popup itself.
2. Sometimes the popup stays after the opt-out completes. This suggests that we're hitting a race condition where we hit the button too fast for the popup to handle it.

File: rules/autoconsent/taunton.json [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  "vendorUrl": "https://www.taunton.com/",
+  "prehideSelectors": ["#taunton-user-consent__overlay"],
+  "detectCmp": [
+    {
+      "exists": "#taunton-user-consent__overlay"
+    }
+  ],
+  "detectPopup": [
+    {
+      "visible": "#taunton-user-consent__overlay"
```
Comment (sammacbeth) [inline rules/autoconsent/taunton.json:12]: This is always considered 'visible' by `isElementVisible`, causing this rule to run every time. They're setting an `aria-hidden:"true"` attribute though, I wonder if we could use that to figure out visibility?

**Review by sammacbeth — CHANGES_REQUESTED** [review]
I had a couple of issues with rules triggering when the popup is not shown. I wonder if we can improve those rules a bit?

File: rules/autoconsent/taunton.json (hunk 3 above)
Comment (muodov) [reply rules/autoconsent/taunton.json:12]: oh, nice catch, thank you! `aria-hidden` works well

File: tests/om.spec.ts (hunk 2 above)
Comment (muodov) [reply tests/om.spec.ts:7]: It was indeed a race condition 🙏 I changed the selectors, and not it seems to be more stable

File: rules/autoconsent/openli.json (hunk 1 above)
Comment (muodov) [reply rules/autoconsent/openli.json]: Thanks for catching this! I made the selectors more specific

**Review by sammacbeth — APPROVED** [review]

**Pull request merged**